internal/venice-common/src/test/java/com/linkedin/venice/writer/VeniceWriterUnitTest.java

```diff
@@ -687,41 +690,34 @@ public void testGlobalDivChunking() {
-      }
+      writer.sendGlobalRtDivMessage(0, "test-key", new String(valueChars));
 
       ArgumentCaptor<KafkaMessageEnvelope> kmeArgumentCaptor = ArgumentCaptor.forClass(KafkaMessageEnvelope.class);
-      ArgumentCaptor<KafkaKey> kafkaKeyArgumentCaptor = ArgumentCaptor.forClass(KafkaKey.class);
-
-      if (size == SMALL_VALUE_SIZE) {
-        // 1 SOS, 1 DivControlMessage
-        verify(mockedProducer, times(2))
-            .sendMessage(any(), any(), kafkaKeyArgumentCaptor.capture(), kmeArgumentCaptor.capture(), any(), any());
-      } else { // TOO_LARGE_VALUE_SIZE
-        // 1 SOS, 4 DivChunk, 1 DivManifest
-        verify(mockedProducer, times(6))
-            .sendMessage(any(), any(), kafkaKeyArgumentCaptor.capture(), kmeArgumentCaptor.capture(), any(), any());
-      }
+      ArgumentCaptor<KafkaKey> keyArgumentCaptor = ArgumentCaptor.forClass(KafkaKey.class);
+
+      // SMALL_VALUE_SIZE: 1 SOS, 1 GlobalRtDiv Message
+      // TOO_LARGE_VALUE_SIZE: 1 SOS, 4 DivChunk, 1 DivManifest
+      final int invocationCount = (size == SMALL_VALUE_SIZE) ? 2 : 6;
+      verify(mockedProducer, times(invocationCount))
+          .sendMessage(any(), any(), keyArgumentCaptor.capture(), kmeArgumentCaptor.capture(), any(), any());
 
-      for (KafkaKey key: kafkaKeyArgumentCaptor.getAllValues()) {
-        Assert.assertTrue(key.isDivControlMessage() || key.isControlMessage());
+      for (KafkaKey key: keyArgumentCaptor.getAllValues()) {
+        assertTrue(key.isGlobalRtDiv() || key.isControlMessage());
       }
 
+      final Set<Integer> validSchemaIds = new HashSet<>(
+          Arrays.asList(
+              AvroProtocolDefinition.CHUNK.getCurrentProtocolVersion(),
+              AvroProtocolDefinition.CHUNKED_VALUE_MANIFEST.getCurrentProtocolVersion(),
+              AvroProtocolDefinition.GLOBAL_RT_DIV_STATE.getCurrentProtocolVersion()));
       for (KafkaMessageEnvelope kme: kmeArgumentCaptor.getAllValues()) {
         if (kme.messageType == MessageType.CONTROL_MESSAGE.getValue()) {
-          Assert.assertTrue(
-              ((ControlMessage) kme.getPayloadUnion()).getControlMessageType() == ControlMessageType.START_OF_SEGMENT
-                  .getValue());
+          ControlMessage controlMessage = ((ControlMessage) kme.getPayloadUnion());
+          assertEquals(ControlMessageType.START_OF_SEGMENT.getValue(), controlMessage.getControlMessageType());
         } else {
-          Assert.assertTrue(kme.messageType == MessageType.PUT.getValue());
           Put put = (Put) kme.payloadUnion;
-          Assert.assertTrue(
-              put.getSchemaId() == AvroProtocolDefinition.CHUNK.getCurrentProtocolVersion()
-                  || put.getSchemaId() == AvroProtocolDefinition.CHUNKED_VALUE_MANIFEST.getCurrentProtocolVersion()
-                  || put.getSchemaId() == AvroProtocolDefinition.GLOBAL_DIV_STATE.getCurrentProtocolVersion());
+          assertEquals(kme.messageType, MessageType.GLOBAL_RT_DIV.getValue());
```

Comment:
Ditto. The KME message type is `PUT` in the PR's description.

Reply:
Fixed.